ingest/src/main/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClient.java

```diff
@@ -39,19 +37,21 @@
  * Since the streaming client communicates directly with the engine, it's more prone to failure, so this class
  * holds both a streaming client and a queued client.
  * It tries {@value ATTEMPT_COUNT} times using the streaming client, after which it falls back to the queued streaming client in case of failure.
- * If the size of the stream is bigger than {@value MAX_STREAMING_SIZE_BYTES}, it will fall back to the queued streaming client.
+ * By default the policy for choosing a queued ingestion on the first try is the checking of weather the size of the estimated
+ * raw stream size (a conversion to compressed CSV) is bigger than 4MB, it will fall back to the queued streaming client.
+ * Use SourceInfo.size to override size estimations, alternatively - use setQueuingPolicy to override the predicate heuristics.
  * <p>
  */
 public class ManagedStreamingIngestClient extends IngestClientBase implements QueuedIngestClient {
 
     private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
     public static final int ATTEMPT_COUNT = 3;
-    public static final int MAX_STREAMING_SIZE_BYTES = 4 * 1024 * 1024;
     public static final String CLASS_NAME = ManagedStreamingIngestClient.class.getSimpleName();
     final QueuedIngestClientImpl queuedIngestClient;
     final StreamingIngestClient streamingIngestClient;
     private final ExponentialRetry exponentialRetryTemplate;
     private CloseableHttpClient httpClient = null;
+    private ManagedStreamingQueuingPolicy queuingPolicy = ManagedStreamingQueuingPolicy.Default;
```

Comment:
Do we want to this here, like this, when v2 is in the horizon? I think we should talk to Yochai.

I don't mean the check itself, but making it customizable for the user.

We are commiting to this interface, which I'm not a big fan of, and I don't know if we have need for this to be supported.

In general, I have two concerns:

1. Why do we have the Policy class if it only has one member, and not just put the member directly
2. Why is the member a functional thing and not an interface?

Reply:
1. talked to Yochai - he doesnt mind people changing it 
2. As its a predicate - thought it is a nice Java impl ... will change to interface 

This class will most likely grow in time

Reply:
I mean it's unlikely to grow if we're going to replace most of it for v2, but sure